arcconfig/arcconfig.go

```diff
@@ -0,0 +1,164 @@
+	Name string `yaml:"name"`
+
+	// Authentication defines the AuthenticationRequirement
+	Authentication AuthenticationRequirement `yaml:"authentication,omitempty"`
+
+	// Authorization defines the ConsumerAuthorization
+	Authorization ConsumerAuthorization `yaml:"authorization,omitempty"`
+
+	// HTTP matching rules for this operation.
+	// The path templates within which should be fully denormalized.
+	HTTPMatches []HTTPMatch `yaml:"http_match,omitempty"`
+
+	// Name of the target server for this operation. This will be sent to Envoy
+	// for routing to the corresponding upstream cluster upon a successful
+	// authorization of the operation.
+	Target string `yaml:"target"`
+}
+
+// AuthenticationRequirement defines the authentication requirement.
+// Precisely one of JWT, Any and All should be set.
+type AuthenticationRequirement struct {
+	// JWT defines the JWTAuthentication
+	JWT JWTAuthentication `yaml:"jwt,omitempty"`
+
+	// Any contains a list of AuthenticationRequirements.
+	// A successful authentication requires one of them being satisfied.
+	Any []AuthenticationRequirement `yaml:"any,omitempty"`
+
+	// All contains a list of AuthenticationRequirements.
+	// A successful authentication requires all of them being satisfied.
+	All []AuthenticationRequirement `yaml:"all,omitempty"`
+}
+
+// JWTAuthentication defines the JWT authentication
+type JWTAuthentication struct {
+	// Name of this JWT Provider, unique within the Proxy.
+	Name string `yaml:"name"`
+
+	// JWT issuer ("iss" claim)
+	Issuer string `yaml:"issuer"`
+
+	// A remote JWKS source
+	RemoteJWKS RemoteJWKS `yaml:"remote_jwks"`
+
+	// Audiences contains a list of audiences
+	Audiences []string `yaml:"audiences,omitempty"`
+
+	// Header name that will contain decoded JWT payload in requests forwarded to
+	// target.
+	ForwardPayloadHeader string `yaml:"forward_payload_header,omitempty"`
+
+	// Locations where JWT may be found. First match wins.
+	In []HTTPParameter `yaml:"in"`
+}
+
+// RemoteJWKS contains information for remote JWKS
+type RemoteJWKS struct {
+	// URL of the JWKS
+	URL string `yaml:"url"`
+
+	// CacheDuration of the JWKS
+	CacheDuration time.Duration `yaml:"cache_duration,omitempty"`
+}
+
+// ConsumerAuthorization is the configuration of API consumer authorization
+type ConsumerAuthorization struct {
+	FailOpen bool `yaml:"fail_open,omitempty"`
+
+	// Locations of API consumer credential (API Key). First match wins.
+	In []HTTPParameter `yaml:"in"`
+
+	// Quota identifier (optional)
+	QuotaIdentifier string `yaml:"quota_identifier,omitempty"`
+}
+
+// HTTPMatch is an HTTP request matching rule
+type HTTPMatch struct {
+	// URL path template using to match incoming requests and optionally identify
+	// path variables. This should be fully denormalized as there is no global
+	// basepath defined anywhere.
+	PathTemplate string `yaml:"path_template"`
+
+	// HTTP method
+	Method HTTPMethod `yaml:"method,omitempty"`
```

Comment:
I see this is an `int` alias, however the YAML form will be a string e.g. `GET`, `POST`, etc. How do we plan on reconciling this?

Reply:
I actually just changed this to string as Scott had the similar comment.